### PR TITLE
feat: Handle app link intent for events

### LINKED ISSUE #1833: Add content provider to receive eventyay event link
**Describe the feature you'd like**
Receive eventyay event link in attendee and display event details. 

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #1833

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 1]
```diff
@@ -135,6 +135,17 @@ class EventsFragment : Fragment(), ScrollToTop {
                 showNoInternetScreen(!isConnected && eventsViewModel.events.value == null)
             })
 
+        eventsViewModel.appLinkOrderId
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: @iamareebjamal @anhanh11001 I need help here. It is not observing the data for which I set value from the main activity. How to pass live data from the activity to events fragment? 
Last opting I got is shared preference, but I don't think it's good to use that.

**Comment by iamareebjamal** [comment]
You have to pass the data in intent Bundle, not in viewmodel

**Comment by liveHarshit** [comment]
Updated

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 2]
```diff
@@ -133,12 +134,15 @@ class EventDetailsFragment : Fragment() {
...
-        if (currentEvent != null)
-            loadEvent(currentEvent)
-        else
-            eventViewModel.loadEvent(safeArgs.eventId)
+        when {
+            currentEvent != null -> loadEvent(currentEvent)
+            !eventIdentifier.isNullOrEmpty()-> eventViewModel.loadEventByIdentifier(eventIdentifier)
+            else -> eventViewModel.loadEvent(safeArgs.eventId)
+        }
+
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: extra

**Pull request merged**